File: Corte1/Java/Ejercicio4/Ejercicio4.java
class Estudiante {
    private String nombre;
    private double nota;

    public Estudiante(String nombre, double nota) {
        this.nombre = nombre;
        setNota(nota);
    }

    public String getNombre() {
        return nombre;
    }

    public double getNota() {
        return nota;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setNota(double nota) {
        if (nota >= 0) {
            this.nota = nota;
        } else {
            System.out.println("Nota inválida");
        }
    }
}

public class Main {
    public static void main(String[] args) {
        Estudiante e = new Estudiante("Juan", 4.5);
        System.out.println(e.getNombre() + " " + e.getNota());

        e.setNota(-2);
        e.setNota(3.8);
        System.out.println(e.getNota());
    }
}

File: Corte1/Python/Ejercicio4/Ejercicio4.py
class Estudiante:
    def __init__(self, nombre, nota):
        self.__nombre = nombre
        self.__nota = 0
        self.set_nota(nota)

    def get_nombre(self):
        return self.__nombre

    def get_nota(self):
        return self.__nota

    def set_nombre(self, nombre):
        self.__nombre = nombre

    def set_nota(self, nota):
        if nota >= 0:
            self.__nota = nota
        else:
            print("Nota inválida")

e = Estudiante("Juan", 4.5)
print(e.get_nombre(), e.get_nota())

e.set_nota(-2)
e.set_nota(3.8)
print(e.get_nota())
